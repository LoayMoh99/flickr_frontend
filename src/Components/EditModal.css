main {
  background: #fff;
  margin: 0 auto;
  max-width: 680px;
  padding: 50px 30px;
}

.modal__backdrop {
  background: rgba(255, 254, 254, 0.705);
  bottom: 0;
  left: 0;
  overflow: auto;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

.modal__container {
  background: #fff;
  border-radius: 5px;
  max-width: 1496px;
  max-height: 416px;
  height: 416px;
  margin: 250px auto;
  padding: 15px;
  width: 600px;
  border-radius: 3px;
  box-shadow: 0 1px 6px rgb(0 0 0 / 20%);
  box-sizing: border-box;
}

.modal__title {
  font-size: 17px;
  padding: 19px 2px 16px;
  color: #5c5f61;
  line-height: 17px;
}

.modal__container p {
  padding: 20px;
  min-height: 60px;
  overflow-y: auto;
  padding-top: 0;
  font-size: 14px;
}

#cancel {
  height: 29px;
  line-height: 29px;
  min-width: 80px;
  font-size: 14px;
  background-color: #979ea2;
  color: #fff;
  box-sizing: border-box;
  display: inline-block;
  border-radius: 3px;
  border: none;
  cursor: pointer;
  margin-left: 10px;
  padding: 0 20px;
  position: relative;
}

#confirm {
  height: 29px;
  line-height: 29px;
  min-width: 80px;
  font-size: 14px;
  background-color: #c82426;
  color: #fff;
  box-sizing: border-box;
  display: inline-block;
  border-radius: 3px;
  border: none;
  cursor: pointer;
  margin-left: 10px;
  padding: 0 20px;
  position: relative;
}

.modal__container button {
  float: right;
}

/* The Close Button */
.close {
  color: #aaaaaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.close:hover,
.close:focus {
  color: #000;
  text-decoration: none;
  cursor: pointer;
}

.title-desc-container {
  width: 560px;
  background: #fff;
  border: 1px solid #9ea4a7;
  border-radius: 3px;
  overflow: hidden;
  box-sizing: border-box;
}
.edit-title {
  margin-top: 0;
  margin-bottom: 0;
  height: 25px;
  width: 100%;
  padding: 8px 10px;
  border-style: hidden;
  background: transparent;
  box-shadow: none;
}
.edit-description {
  margin-top: 0;
  margin-bottom: 0;
  height: 48px;
  width: 100%;
  padding: 8px 10px;
  border-style: none;
  background: transparent;
  box-shadow: none;
}
.textarea {
  -webkit-writing-mode: horizontal-tb !important;
  text-rendering: auto;
  color: -internal-light-dark(black, white);
  letter-spacing: normal;
  word-spacing: normal;
  text-transform: none;
  text-indent: 0px;
  text-shadow: none;
  display: inline-block;
  text-align: start;
  appearance: auto;
  background-color: -internal-light-dark(rgb(255, 255, 255), rgb(59, 59, 59));
  -webkit-rtl-ordering: logical;
  flex-direction: column;
  resize: auto;
  cursor: text;
  white-space: pre-wrap;
  overflow-wrap: break-word;
  column-count: initial !important;
  margin: 0em;
  font: 400 13.3333px Arial;
  border-width: 1px;
  border-style: solid;
  border-color: -internal-light-dark(rgb(118, 118, 118), rgb(133, 133, 133));
  border-image: initial;
  padding: 2px;
}

.privacy-label {
  list-style: none;
  margin-top: 10px;
  color: #898989;
  font-size: 14px;
  font-weight: 400;
  max-width: 325px;
  display: inline-block;
  line-height: 22px;
}

.truncate {
  max-width: 205px;
  display: inline-block;
  vertical-align: baseline;
  font-weight: 400;
  color: #006dac;
  font-size: 14px;
  padding-left: 10px;
}

.fluid .modal .no-outline,
.fluid .modal label:hover {
  outline-width: 0;
}

.ui-dropdown > span {
  text-align: left;
  color: #fff;
  font-size: 1em;
  white-space: nowrap;
  padding-right: 16px;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAGCAQAAABQ+cdNAAAAOUlEQVQIW2P4L/lfgwEJ/Nf4L8fwX/f/s//mcCFzIM8YxHCCCYOFnGDyYGEUIbgwqhBY2BhsFhgAAGa6JziJ9SClAAAAAElFTkSuQmCC);
  background-color: transparent;
  background-repeat: no-repeat;
  background-position: 100%;
  cursor: pointer;
  z-index: 2002;
}
