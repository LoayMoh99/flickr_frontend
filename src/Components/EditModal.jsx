/* eslint-disable linebreak-style */
/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable linebreak-style */
/* eslint-disable max-len */
/* eslint-disable linebreak-style */
import React from 'react';
import './EditModal.css';

function EditModal(props) {
  return (
    <>

      {/* <button type="button" className="btn btn-primary" data-toggle="modal" data-target="#exampleModalCenter">
        Launch demo modal
      </button>

      // eslint-disable-next-line max-len
      <div className="modal" id="exampleModalCenter" tabIndex="-1" role="dialog" aria-labelledby="exampleModalCenterTitle" aria-hidden="true">
        <div className="modal-dialog modal-dialog-centered" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title" id="exampleModalLongTitle">Modal title</h5>
              <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              ...
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-secondary" data-dismiss="modal">Close</button>
              <button type="button" className="btn btn-primary">Save changes</button>
            </div>
          </div>
        </div>
      </div> */}

      <div className="modal__backdrop" onClick={props.onRequestClose}>
        <div className="modal__container">
          <span className="close" onClick={props.onRequestClose}>&times;</span>
          <h3 className="modal__title">Editing # photos </h3>
          <div className="title-desc-container">
            <input className="edit-title" type="text" placeholder="Change title" />
            <hr />
            <textarea className="edit-description" placeholder="Change description" />

          </div>
          <span className="privacy-label">Who can see this photo ?</span>
          <span className="truncate no-outline" data-action="toggle" aria-haspopup="true" aria-expanded="false">Friends & Family </span>
          <li data-action="set" data-value="public" data-text="Public">Public</li>
        </div>
      </div>

    </>
  );
}

export default EditModal;
