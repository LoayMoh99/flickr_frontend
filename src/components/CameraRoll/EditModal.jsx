/* eslint-disable linebreak-style */
/* eslint-disable react/prop-types */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable linebreak-style */
/* eslint-disable max-len */
/* eslint-disable linebreak-style */
import React, {useState} from 'react';
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// import { fa-tag } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTag } from '@fortawesome/free-solid-svg-icons';
import { faUserFriends } from '@fortawesome/free-solid-svg-icons';
import './EditModal.css';
import DropdownPrivacy from './DropdownPrivacy'; 

function EditModal(props) {
  const tag = <FontAwesomeIcon icon={faTag} color="DarkGrey" />;
  const people = <FontAwesomeIcon icon={faUserFriends} color="DarkGrey" />;
  const [tagAdded,setTagAdded] = useState(false);
  const [peopleAdded, setPeopleAdded]= useState(false);
  const [inputTag, setInputTag] = useState("");
  const [inputPeople , setInputPeople] = useState("");
  
  function handleTagChange(event){
    setTagAdded(true);
    const newTag = event.target.value;
    setInputTag(newTag);
    console.log(inputTag);
    if(newTag==="")
    {
      setTagAdded(false);
    }
  }
  function handlePeopleChange(event){
    setPeopleAdded(true);
    const newPeople = event.target.value;
    setInputPeople(newPeople);
    console.log(inputPeople);
    if(newPeople==="")
    {
      setPeopleAdded(false);
    }
  }

  


  return (
    <>

      <div className="modal__backdrop_edit">
        <div className="modal__container_edit">
          <span className="close_edit" onClick={props.onRequestEditClose}>&times;</span>
          <h3 className="modal__title_edit">
            Editing
            {' '}
            {props.countEdit}
            {' '}
            photos
            {' '}
          </h3>
          <div className="title-desc-container">
            <input className="edit-title" type="text" placeholder="Change title" onChange={handleTagChange}/>
            <hr />
            <textarea className="edit-description" placeholder="Change description" onChange={handlePeopleChange}/>

          </div>
          <span className="privacy-label">Who can see this photo ?</span>

          <span className="space" />
          <DropdownPrivacy />
          <div className="row">
            <div className="edit-tag">
              {tag}
              {'  '}
              <input type="text" className="no-outline" placeholder="Add new tags" />
            </div>
            <div className="edit-tag">
              {people}
              {'  '}
              <input type="text" className="no-outline" placeholder="Add new people" />
            </div>
          </div>
          <div className="edit-footer">
            <button
              id="save_edit"
              type="button"
              onClick={props.onRequestEditClose} // should be modified to actually upadate the database with the last edit
            >
              Save
            </button>
            <button id="cancel_edit" type="button" onClick={props.onRequestEditClose}>
              Cancel
            </button>
          </div>
        </div>
      </div>

    </>
  );
}

export default EditModal;
