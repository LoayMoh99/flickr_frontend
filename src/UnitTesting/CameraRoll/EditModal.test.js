import React from 'react';
import ReactDOM from 'react-dom';
import EditModal from '../../components/CameraRoll/EditModal';
import Enzyme, { shallow,mount } from 'enzyme';
import {render, fireEvent, cleanup, getByPlaceholderText} from '@testing-library/react';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

afterEach(cleanup)

it('EditModal renders without crashing', ()=>{
    const dummy =()=>{return 1;};
    const div = document.createElement('div');
    ReactDOM.render(<EditModal
        onRequestEditClose={dummy}
        imgEdit="[{_id:1},{_id:2},{_id:3}]"
        imgEditIds= "[1,2,3]"
        countEdit="3"
        />, div);

});

it('EditModal renders correctly enzyme', () => {
    const dummy =()=>{return 1;};
    const wrapper = shallow(<EditModal
        onRequestEditClose={dummy}
        imgEdit="[{_id:1},{_id:2},{_id:3}]"
        imgEditIds= "[1,2,3]"
        countEdit="3"
        />)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  it('Inputing photo title updates the state', () => {
    const dummy =()=>{return 1;};
    const { getByPlaceholderText} = render(<EditModal
        onRequestEditClose={dummy}
        imgEdit="[{_id:1},{_id:2},{_id:3}]"
        imgEditIds= "[1,2,3]"
        countEdit="3"
        />);

    expect(getByPlaceholderText("Change title").value).toBe("")

    fireEvent.change(getByPlaceholderText("Change title"), {target: {value: 'Text' } } )

    expect(getByPlaceholderText("Change title").value).not.toBe("")
 })

 it('Inputing photo description updates the state', () => {
    const dummy =()=>{return 1;};
    const { getByPlaceholderText} = render(<EditModal
        onRequestEditClose={dummy}
        imgEdit="[{_id:1},{_id:2},{_id:3}]"
        imgEditIds= "[1,2,3]"
        countEdit="3"
        />);

    expect(getByPlaceholderText("Change description").value).toBe("")

    fireEvent.change(getByPlaceholderText("Change description"), {target: {value: 'Text' } } )

    expect(getByPlaceholderText("Change description").value).not.toBe("")
 })

 it('Inputing new tags updates the state', () => {
    const dummy =()=>{return 1;};
    const { getByPlaceholderText} = render(<EditModal
        onRequestEditClose={dummy}
        imgEdit="[{_id:1},{_id:2},{_id:3}]"
        imgEditIds= "[1,2,3]"
        countEdit="3"
        />);

    expect(getByPlaceholderText("Add new tags").value).toBe("")

    fireEvent.change(getByPlaceholderText("Add new tags"), {target: {value: 'Text' } } )

    expect(getByPlaceholderText("Add new tags").value).not.toBe("")
 })

 it('Inputing new people updates the state', () => {
    const dummy =()=>{return 1;};
    const { getByPlaceholderText} = render(<EditModal
        onRequestEditClose={dummy}
        imgEdit="[{_id:1},{_id:2},{_id:3}]"
        imgEditIds= "[1,2,3]"
        countEdit="3"
        />);

    expect(getByPlaceholderText("Add new people").value).toBe("")

    fireEvent.change(getByPlaceholderText("Add new people"), {target: {value: 'Text' } } )

    expect(getByPlaceholderText("Add new people").value).not.toBe("")
 })
