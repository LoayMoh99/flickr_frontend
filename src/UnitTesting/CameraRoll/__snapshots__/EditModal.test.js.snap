// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditModal renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="modal__backdrop_edit"
  >
    <div
      className="modal__container_edit"
    >
      <span
        className="close_edit"
        onClick={[Function]}
      >
        Ã—
      </span>
      <h3
        className="modal__title_edit"
      >
        Editing
         
        3
         
        photos
         
      </h3>
      <div
        className="title-desc-container"
      >
        <input
          className="edit-title"
          onChange={[Function]}
          placeholder="Change title"
          type="text"
        />
        <hr />
        <textarea
          className="edit-description"
          onChange={[Function]}
          placeholder="Change description"
        />
      </div>
      <span
        className="privacy-label"
      >
        Who can see this photo ?
      </span>
      <span
        className="space"
      />
      <DropdownPrivacy
        imgEditIds="[1,2,3]"
      />
      <div
        className="row"
      >
        <div
          className="edit-tag"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            color="DarkGrey"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f02b",
                  "M0 252.118V48C0 21.49 21.49 0 48 0h204.118a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882L293.823 497.941c-18.745 18.745-49.137 18.745-67.882 0L14.059 286.059A48 48 0 0 1 0 252.118zM112 64c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48z",
                ],
                "iconName": "tag",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
            
          <input
            className="no-outline"
            onChange={[Function]}
            placeholder="Add new tags"
            type="text"
          />
        </div>
        <div
          className="edit-tag"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            color="DarkGrey"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  640,
                  512,
                  Array [],
                  "f500",
                  "M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z",
                ],
                "iconName": "user-friends",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
            
          <input
            className="no-outline"
            onChange={[Function]}
            placeholder="Add new people"
            type="text"
          />
        </div>
      </div>
      <div
        className="edit-footer"
      >
        <button
          id="save_edit"
          onClick={[Function]}
          type="button"
        >
          Save
        </button>
        <button
          id="cancel_edit"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</Fragment>
`;
